src/math.rs

```diff
@@ -12,18 +12,18 @@ use super::share::Share;
 // The expected `shares` argument format is the same as the output by the `get_evaluator´ function.
 // Where each (key, value) pair corresponds to one share, where the key is the `x` and the value is a vector of `y`,
 // where each element corresponds to one of the secret's byte chunks.
-pub fn interpolate<const POLY: u16>(shares: &[Share<POLY>]) -> Vec<u8> {
-    (0..shares[0].y.len())
+pub fn interpolate<const POLY: u16>(shares: &[(GF256<POLY>, Share<POLY>)]) -> Vec<u8> {
```

Comment:
Is it worth using a named tuple struct in here now?

Reply:
removed this tuple, and using Share (with x-coordinate) directly.